jar-infer-lib/src/main/java/com/uber/nullaway/jarinfer/DefinitelyDerefedParamsDriver.java

```diff
@@ -0,0 +1,398 @@
+    System.out.println("extracting " + aarPath + "...");
+    String classDir = FilenameUtils.getFullPath(aarPath) + FilenameUtils.getBaseName(aarPath);
+    try {
+      JarFile aar = new JarFile(aarPath);
+      Enumeration enumEntries = aar.entries();
+      while (enumEntries.hasMoreElements()) {
+        JarEntry aarEntry = (JarEntry) enumEntries.nextElement();
+        if (aarEntry.getName().endsWith("classes.jar")) {
+          JarInputStream jis = new JarInputStream(aar.getInputStream(aarEntry));
+          JarEntry jarEntry = null;
+          while ((jarEntry = jis.getNextJarEntry()) != null) {
+            if (jarEntry.getName().endsWith(DEFAULT_ASTUBX_LOCATION)) {
+              throw new Error("jar-infer called on already processed library: " + aarPath);
+            }
+            if (jarEntry.isDirectory() || !jarEntry.getName().endsWith(".class")) continue;
+            File f = new File(classDir + File.separator + jarEntry.getName());
+            f.getParentFile().mkdirs();
+            FileOutputStream fos = new FileOutputStream(f);
+            IOUtils.copy(jis, fos);
+            fos.close();
+          }
+          jis.close();
+        }
+      }
+      aar.close();
+    } catch (IOException e) {
+      throw new Error(e);
+    }
+    return classDir;
+  }
+
+  /**
+   * Write processed jar with nullability model in jar -> META-INF/nullaway/jarinfer.astubx
+   *
+   * @param inJarPath Path of input jar file.
+   * @param outJarPath Path of output jar file.
+   * @param map_mtd_result Map of 'method references' to their 'list of NonNull parameters'.
+   */
+  private static void writeProcessedJAR(
+      String inJarPath, String outJarPath, Result map_mtd_result) {
+    Preconditions.checkArgument(
+        inJarPath.endsWith(".jar") && Files.exists(Paths.get(inJarPath)),
+        "invalid jar file! " + inJarPath);
+    try {
+      JarFile jar = new JarFile(inJarPath);
+      JarOutputStream jos = new JarOutputStream(new FileOutputStream(outJarPath));
+      Enumeration enumEntries = jar.entries();
+      while (enumEntries.hasMoreElements()) {
+        JarEntry jarEntry = (JarEntry) enumEntries.nextElement();
+        InputStream jis = jar.getInputStream(jarEntry);
+        jos.putNextEntry(jarEntry);
+        IOUtils.copy(jis, jos);
+        jis.close();
+      }
+      jos.putNextEntry(new JarEntry(DEFAULT_ASTUBX_LOCATION));
+      writeModel(new DataOutputStream(jos), map_mtd_result);
+      jos.close();
+      jar.close();
+      System.out.println("processed jar to: " + outJarPath);
+    } catch (IOException e) {
+      throw new Error(e);
+    }
+  }
+
+  /**
+   * Write processed aar with nullability model in aar -> classes.jar ->
+   * META-INF/nullaway/jarinfer.astubx
+   *
+   * @param inAarPath Path of input aar file.
+   * @param outAarPath Path of output aar file.
+   * @param map_mtd_result Map of 'method references' to their 'list of NonNull parameters'.
+   */
+  private static void writeProcessedAAR(
```

Comment:
Again there seems to be some code duplication between this and `writeProcessedJAR`.